test/test_dagnav.py

```diff
@@ -743,3 +851,137 @@ def test_surface_load_file(request):
+    assert len(initial_parents) == 0, "Surface should have no parents initially"
+
+    # Test setting forward=None, reverse=vol1
+    surf.senses = [None, vol1]
+    assert surf.forward_volume is None
+    assert surf.reverse_volume == vol1
+
+    # Check parents: vol1 should be added
+    parents_after_none_vol1 = model.mb.get_parent_meshsets(surf.handle)
+    assert set(parents_after_none_vol1) == {vol1.handle}
+    assert len(parents_after_none_vol1) == 1
+
+    # Test setting forward=vol1, reverse=None
+    surf.senses = [vol1, None]
+    assert surf.forward_volume == vol1
+    assert surf.reverse_volume is None
+
+    # Check parents: vol1 should still be the only parent
+    parents_after_vol1_none = model.mb.get_parent_meshsets(surf.handle)
+    assert set(parents_after_vol1_none) == {vol1.handle}
+    assert len(parents_after_vol1_none) == 1
+
+    # Test setting forward=vol2, reverse=vol1
+    surf.senses = [vol2, vol1]
+    assert surf.forward_volume == vol2
+    assert surf.reverse_volume == vol1
+    parents_after_vol2_vol1 = model.mb.get_parent_meshsets(surf.handle)
+    assert set(parents_after_vol2_vol1) == {vol1.handle, vol2.handle}
+    assert len(parents_after_vol2_vol1) == 2
+
+    # Test setting both=None
+    # Crucially, the current senses setter *does not remove* existing parents.
+    # Therefore, vol1 and vol2 handles remain parents.
+    surf.senses = [None, None]
+    assert surf.forward_volume is None
+    assert surf.reverse_volume is None
+
+    # Check parents: Parent handles persist because the setter only adds, never removes.
+    parents_after_none_none = model.mb.get_parent_meshsets(surf.handle)
+    expected_parents_at_end = {vol1.handle, vol2.handle}
+    assert set(parents_after_none_none) == expected_parents_at_end
+    assert len(parents_after_none_none) == len(expected_parents_at_end), \
+        f"Expected parents {expected_parents_at_end} to persist"
+
+    # Test setting back to a valid sense
+    # Ensure we can re-establish sense and parent links correctly
+    surf.senses = [vol1, vol2]
+    assert surf.forward_volume == vol1
+    assert surf.reverse_volume == vol2
+    parents_final = model.mb.get_parent_meshsets(surf.handle)
+
+    # Should still be vol1 and vol2 (re-added harmlessly)
+    assert set(parents_final) == {vol1.handle, vol2.handle}
+    assert len(parents_final) == 2
+
+def test_surface_create_invalid_filename():
+    """Test create_surface with a non-STL file."""
+    model = pydagmc.Model()
+    with pytest.raises(ValueError, match="Only STL files are supported"):
+        model.create_surface(filename='my_model.step')
+
+def test_geometryset_category_runtime_error(request):
+    """Test category returns None when tag is missing."""
+    model = pydagmc.Model()
+    raw_handle = model.mb.create_meshset()
+    model.mb.tag_set_data(model.geom_dimension_tag, raw_handle, 2)
+    try:
+        model.mb.tag_delete(model.category_tag, (raw_handle,))
+    except RuntimeError:
```

Comment:
Same same as name tags and sense tags.

Reply:
Maybe a good alternative is to make a test that creates a `Model` object and queries the tags with PyMOAB directly to ensure they exist from the start.